airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java

```diff
@@ -163,6 +167,28 @@ public Process create(final String jobId,
     }
   }
 
+  private static String createContainerName(final String fullImagePath, final String jobId, final int attempt) {
+    final var versionDelimiter = ":";
+    final var noVersion = fullImagePath.split(versionDelimiter)[0];
+
+    final var dockerDelimiter = "/";
+    final var nameParts = noVersion.split(dockerDelimiter);
+    var imageName = nameParts[nameParts.length - 1];
+
+    final var randSuffix = RandomStringUtils.randomAlphabetic(5).toLowerCase();
```

Comment:
Just curious, is the random suffix only necessary due to the fact that we call everything `sync`? So without the random suffix, we'd potentially have name conflicts between check/discover/etc?

Reply:
Not really. The random suffix is there to make it truly random without any assumptions on the job-id/attempt-id combination e.g. that we have globally unique job-id/attempt-id combinations.

The synchronous jobs such as check, discover etc are assigned a job id that is a UUID so we can assume the names will always be random.

Reply:
Does that make sense? Happy to chat more on Slack.